### ISSUE TITLE
Provide consistent NodeAffinity or NodeSelector UX for components

### ISSUE THREAD

**Issue by surajssd** [kinvolk/lokomotive#435]
### Problem

Right now some components support `node_selector` and others support `node_affinity`. It is very incosistent. 

Historically this has been the case because Lokoctl exposed whatever underlying Helm Chart is exposing. Lokoctl never did any magical tranformation from `node_affinity` (at lokoctl) to `node_selector` (at the chart values level).

Also converting from `node_affinity` to `node_selector` is a lossy change because `node_affinity` has so much information that cannot be converted to `node_selector`. But reverse is not true. We can very easily convert from `node_selector` to `node_affinity`, though.

### Proposed solution

We expose only `node_selector` to lokoctl user and depending on what is needed by the chart we do the conversion or just use it as it is.

**Comment by invidian** [comment]
>We expose only node_selector to lokoctl user and depending on what is needed by the chart we do the conversion or just use it as it is.

According to the [documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity), node affinity is an expansion of node selector, so shouldn't we support `node_affinity` instead of `node_selector`?

**Comment by johananl** [comment]
Yes, I think we can drop node selector support. Whatever we do, it should be consistent across components.
